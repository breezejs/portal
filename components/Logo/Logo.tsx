import React, {memo, ReactNode} from 'react';
import styled from 'styled-components';
import Typography from '../Typography';

const StyledTypography = styled(Typography)`
  margin-bottom: 0;
`;

interface ILogoProps {
  children: ReactNode;
}

function Logo ({children}: ILogoProps) {
  return (
    <StyledTypography component='h1' variant='h4'>{children}</StyledTypography>
  );
}

export default memo(Logo);
