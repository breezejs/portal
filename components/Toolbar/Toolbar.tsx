import {darken} from 'polished';
import React, {memo, ReactNode} from 'react';
import styled from 'styled-components';
import {GlobalStylesTheme} from '../../core/GlobalStyles';

const secondaryBg = darken(0.02, GlobalStylesTheme.light);
const Header = styled.header`
  align-items: center;
  background-color: ${GlobalStylesTheme.light};
  background-image: linear-gradient(
    45deg,
    ${GlobalStylesTheme.light} 25%,
    ${secondaryBg} 25%,
    ${secondaryBg} 50%,
    ${GlobalStylesTheme.light} 50%,
    ${GlobalStylesTheme.light} 75%,
    ${secondaryBg} 75%,
    ${secondaryBg} 100%
  );
  background-size: 28.28px 28.28px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: .5rem 1rem;
  position: relative;
  user-select: none;
`;

interface IToolbarProps {
  children?: ReactNode;
}

function Toolbar ({children}: IToolbarProps) {
  return (
    <Header>{children}</Header>
  );
}

export default memo(Toolbar);
