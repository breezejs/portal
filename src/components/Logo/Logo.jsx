import PropTypes from 'prop-types';
import React, {memo} from 'react';
import styled from 'styled-components';
import Typography from '../Typography';

const StyledTypography = styled(Typography)`
  margin-bottom: 0;
`;

function Logo ({children}) {
  return (
    <StyledTypography component='h1' variant='h4'>{children}</StyledTypography>
  );
}

Logo.propTypes = {
  children: PropTypes.string.isRequired
};

export default memo(Logo);
