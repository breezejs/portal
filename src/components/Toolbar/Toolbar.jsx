import {darken} from 'polished';
import PropTypes from 'prop-types';
import React, {memo} from 'react';
import styled from 'styled-components';
import {GlobalStylesTheme} from '../GlobalStyles';

function secondaryBg (theme) {
  return darken(0.02, GlobalStylesTheme[theme]);
}

const Header = styled.header`
  ${({theme}) => `
    align-items: center;
    background-color: ${GlobalStylesTheme[theme]};
    background-image: linear-gradient(
      45deg,
      ${GlobalStylesTheme[theme]} 25%,
      ${secondaryBg(theme)} 25%,
      ${secondaryBg(theme)} 50%,
      ${GlobalStylesTheme[theme]} 50%,
      ${GlobalStylesTheme[theme]} 75%,
      ${secondaryBg(theme)} 75%,
      ${secondaryBg(theme)} 100%
    );
    background-size: 28.28px 28.28px;
    color: ${theme === 'light' ? GlobalStylesTheme.bodyColour : GlobalStylesTheme.white}
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    padding: .5rem ${GlobalStylesTheme.gridGutter};
    position: relative;
    user-select: none;
  `}
`;

function Toolbar ({children, theme}) {
  return (
    <Header theme={theme}>{children}</Header>
  );
}

Toolbar.propTypes = {
  children: PropTypes.element,
  theme: PropTypes.oneOf(['light', 'dark'])
};

Toolbar.defaultProps = {
  children: null,
  theme: 'light'
};

export default memo(Toolbar);
