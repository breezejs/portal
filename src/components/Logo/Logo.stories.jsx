import {text, withKnobs} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react';
import React from 'react';
import GlobalStyles from '../GlobalStyles';
import Logo from './Logo';

const stories = storiesOf('Logo', module);

stories
  .addDecorator(withKnobs)
  .addDecorator((storyFn) => (
    <div>
      <GlobalStyles />
      {storyFn()}
    </div>
  ))
  .add('with brand text', () => (
    <Logo>{text('Brand', 'BreezeJS')}</Logo>
  ));
