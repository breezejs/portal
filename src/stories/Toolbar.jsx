import {text, withKnobs} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react';
import React from 'react';
import GlobalStyles from '../components/GlobalStyles';
import Logo from '../components/Logo';
import Toolbar from '../components/Toolbar';

const stories = storiesOf('Toolbar', module);

stories
  .addDecorator(withKnobs)
  .addDecorator((storyFn) => (
    <div>
      <GlobalStyles />
      {storyFn()}
    </div>
  ))
  .add('without content', () => (
    <Toolbar />
  ))
  .add('with content', () => (
    <Toolbar>
      <Logo>{text('Logo text', 'BreezeJS')}</Logo>
    </Toolbar>
  ));
