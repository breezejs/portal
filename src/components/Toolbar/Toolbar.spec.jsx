import {mount, shallow} from 'enzyme';
import React from 'react';
import Logo from '../Logo';
import Toolbar from './Toolbar';

describe('Toolbar', () => {
  describe('with default theme', () => {
    it('should render', () => {
      const result = shallow(
        <Toolbar />
      );

      expect(result).toMatchSnapshot();
    });

    it('should render with a child components', () => {
      const result = mount(
        <Toolbar>
          <Logo>Test</Logo>
        </Toolbar>
      );

      expect(result).toMatchSnapshot();
    });

    it('should have the correct css styles', () => {
      const result = mount(
        <Toolbar />
      );

      expect(result.find('header')).toMatchSnapshot();
    });
  });

  describe('with light theme', () => {
    it('should render', () => {
      const result = shallow(
        <Toolbar theme='light' />
      );

      expect(result).toMatchSnapshot();
    });

    it('should render with a child components', () => {
      const result = mount(
        <Toolbar theme='light'>
          <Logo>Test</Logo>
        </Toolbar>
      );

      expect(result).toMatchSnapshot();
    });

    it('should have the correct css styles', () => {
      const result = mount(
        <Toolbar theme='light' />
      );

      expect(result.find('header')).toMatchSnapshot();
    });
  });

  describe('with dark theme', () => {
    it('should render', () => {
      const result = shallow(
        <Toolbar theme='dark' />
      );

      expect(result).toMatchSnapshot();
    });

    it('should render with a child components', () => {
      const result = mount(
        <Toolbar theme='dark'>
          <Logo>Test</Logo>
        </Toolbar>
      );

      expect(result).toMatchSnapshot();
    });

    it('should have the correct css styles', () => {
      const result = mount(
        <Toolbar theme='dark' />
      );

      expect(result.find('header')).toMatchSnapshot();
    });
  });
});
