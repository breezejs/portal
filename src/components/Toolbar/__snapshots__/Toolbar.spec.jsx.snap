// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar with dark theme should have the correct css styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #343a40;
  background-image: linear-gradient( 45deg, #343a40 25%, #2f353a 25%, #2f353a 50%, #343a40 50%, #343a40 75%, #2f353a 75%, #2f353a 100% );
  background-size: 28.28px 28.28px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: .5rem 1rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<header
  className="c0"
/>
`;

exports[`Toolbar with dark theme should render 1`] = `
<styled.header
  theme="dark"
/>
`;

exports[`Toolbar with dark theme should render with a child components 1`] = `
.c1 {
  margin-bottom: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #343a40;
  background-image: linear-gradient( 45deg, #343a40 25%, #2f353a 25%, #2f353a 50%, #343a40 50%, #343a40 75%, #2f353a 75%, #2f353a 100% );
  background-size: 28.28px 28.28px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: .5rem 1rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<Toolbar
  theme="dark"
>
  <styled.header
    theme="dark"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.header",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "header",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme="dark"
    >
      <header
        className="c0"
      >
        <Logo>
          <StyledComponent
            component="h1"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "compare": null,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  margin-bottom: 0;
",
                  ],
                },
                "displayName": "Styled(Component)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                },
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            variant="h4"
          >
            <Typography
              className="c1"
              component="h1"
              variant="h4"
            >
              <styled.h1
                className="c1"
                component="h1"
                variant="h4"
              >
                <StyledComponent
                  className="c1"
                  component="h1"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
    ",
                          [Function],
                          "
  ",
                        ],
                      },
                      "displayName": "styled.h1",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "h1",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  variant="h4"
                >
                  <h1
                    className="c1 c2"
                  >
                    Test
                  </h1>
                </StyledComponent>
              </styled.h1>
            </Typography>
          </StyledComponent>
        </Logo>
      </header>
    </StyledComponent>
  </styled.header>
</Toolbar>
`;

exports[`Toolbar with default theme should have the correct css styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f8f9fa;
  background-image: linear-gradient( 45deg, #f8f9fa 25%, #f2f4f6 25%, #f2f4f6 50%, #f8f9fa 50%, #f8f9fa 75%, #f2f4f6 75%, #f2f4f6 100% );
  background-size: 28.28px 28.28px;
  color: #212529;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: .5rem 1rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<header
  className="c0"
/>
`;

exports[`Toolbar with default theme should render 1`] = `<styled.header />`;

exports[`Toolbar with default theme should render with a child components 1`] = `
.c1 {
  margin-bottom: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f8f9fa;
  background-image: linear-gradient( 45deg, #f8f9fa 25%, #f2f4f6 25%, #f2f4f6 50%, #f8f9fa 50%, #f8f9fa 75%, #f2f4f6 75%, #f2f4f6 100% );
  background-size: 28.28px 28.28px;
  color: #212529;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: .5rem 1rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<Toolbar>
  <styled.header
    theme="light"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.header",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "header",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme="light"
    >
      <header
        className="c0"
      >
        <Logo>
          <StyledComponent
            component="h1"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "compare": null,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  margin-bottom: 0;
",
                  ],
                },
                "displayName": "Styled(Component)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                },
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            variant="h4"
          >
            <Typography
              className="c1"
              component="h1"
              variant="h4"
            >
              <styled.h1
                className="c1"
                component="h1"
                variant="h4"
              >
                <StyledComponent
                  className="c1"
                  component="h1"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
    ",
                          [Function],
                          "
  ",
                        ],
                      },
                      "displayName": "styled.h1",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": "h1",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  variant="h4"
                >
                  <h1
                    className="c1 c2"
                  >
                    Test
                  </h1>
                </StyledComponent>
              </styled.h1>
            </Typography>
          </StyledComponent>
        </Logo>
      </header>
    </StyledComponent>
  </styled.header>
</Toolbar>
`;

exports[`Toolbar with light theme should have the correct css styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f8f9fa;
  background-image: linear-gradient( 45deg, #f8f9fa 25%, #f2f4f6 25%, #f2f4f6 50%, #f8f9fa 50%, #f8f9fa 75%, #f2f4f6 75%, #f2f4f6 100% );
  background-size: 28.28px 28.28px;
  color: #212529;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: .5rem 1rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<header
  className="c0"
/>
`;

exports[`Toolbar with light theme should render 1`] = `
<styled.header
  theme="light"
/>
`;

exports[`Toolbar with light theme should render with a child components 1`] = `
.c1 {
  margin-bottom: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f8f9fa;
  background-image: linear-gradient( 45deg, #f8f9fa 25%, #f2f4f6 25%, #f2f4f6 50%, #f8f9fa 50%, #f8f9fa 75%, #f2f4f6 75%, #f2f4f6 100% );
  background-size: 28.28px 28.28px;
  color: #212529;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: .5rem 1rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  font-size: 1.5rem;
  color: inherit;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.2;
  margin-bottom: .5rem;
}

<Toolbar
  theme="light"
>
  <styled.header
    theme="light"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.header",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "header",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme="light"
    >
      <header
        className="c0"
      >
        <Logo>
          <StyledComponent
            component="h1"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "compare": null,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  margin-bottom: 0;
",
                  ],
                },
                "displayName": "Styled(Component)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                },
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            variant="h4"
          >
            <Typography
              className="c1"
              component="h1"
              variant="h4"
            >
              <styled.h1
                className="c1"
                component="h1"
                variant="h4"
              >
                <StyledComponent
                  className="c1"
                  component="h1"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
    ",
                          [Function],
                          "
  ",
                        ],
                      },
                      "displayName": "styled.h1",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "h1",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  variant="h4"
                >
                  <h1
                    className="c1 c2"
                  >
                    Test
                  </h1>
                </StyledComponent>
              </styled.h1>
            </Typography>
          </StyledComponent>
        </Logo>
      </header>
    </StyledComponent>
  </styled.header>
</Toolbar>
`;
